app/api/todos/[id]/route.ts

```diff
@@ -0,0 +1,90 @@
+import { NextResponse } from "next/server";
+import connectDB from "@/lib/mongodb";
+import Todo from "../../models/todos-model";
+import { TodoCardInfo } from "@/types";
+
+export async function DELETE(
+  request: Request,
+  { params }: { params: { id: string } }
+) {
+  try {
+    await connectDB();
+
+    const { id } = params;
+
+    if (!id) {
+      return NextResponse.json({ error: "ID is required" }, { status: 400 });
+    }
+
+    const result = await Todo.findOneAndDelete({ id: id });
+
+    if (!result) {
+      return NextResponse.json({ error: "Todo not found" }, { status: 404 });
+    }
+
+    return NextResponse.json(
+      {
+        message: "Todo deleted successfully",
+      },
+      { status: 200 }
+    );
+  } catch (e) {
+    return NextResponse.json(
+      { error: `Failed to delete the todo: ${e}` },
```

Comment:
We don't want to send error and stack trace in response as it may be security issue. Better to log it in server.